### PR TITLE
(2377) Admin invites a user

### PR DESCRIPTION
**Pull request by rgarner**
## Changes in this PR
- Admins can invite users
- Users receive the welcome email with a link to reset their password

### CODE REVIEW

**Comment by CristinaRO** [comment]
[Admin invites a user [5]](https://trello.com/c/JQXINuhU/2377-admin-invites-a-user-5)

File: spec/services/create_user_spec.rb
Comment (dragon-dxw) [inline spec/services/create_user_spec.rb:21]: Feels mildly wierd to depend on the new user being the last one -- seems fine, though.

**Review by dragon-dxw — APPROVED** [review]
Might want to wait for someone more qualified to review, but lgtm.

File: spec/services/create_user_spec.rb
Comment (CristinaRO) [reply spec/services/create_user_spec.rb:21]: It felt weird to me too! We can rely on the testing env being artificial conditions though, where we know that's the only email that's going to be sent.

File: Gemfile.lock
Comment (james) [inline Gemfile.lock:183]: What's going on with this commit? Something feels like its gone wrong?

Comment (CristinaRO) [reply Gemfile.lock]: Fixed it!

Comment (rgarner) [reply Gemfile.lock]: @CristinaRO @james worth noting that this *line* itself is not an error; it's pretty helpful. If we lose the `-x76_64-darwin` line, `bundle install` has to invoke the compiler for `libv8-node` instead of picking a pre-compiled platform-specific version for development.

Consequently, my machine has been compiling it for the last half-hour after pulling `develop`. I can't tell if @james was referring to the `RED` commit that this was in somehow having found its way onto this branch after rebase (because Github has no good way of commenting on a commit without referring to some file content) and @CristinaRO misunderstood and fixed the line @james was forced to refer to!

Comment (rgarner) [reply Gemfile.lock]: We fixed this by finding the commit where we lost this line 183 on `devise-authentication`, rebasing interactively and editing that commit and running `git checkout HEAD -- Gemfile.lock && bundle install`.

This was caused by `rm Gemfile.lock && bundle install` recreating the Gemfile.lock without platform information. Bundler itself suggests the solution if you try and `bundle install` while a `Gemfile.lock` contains conflicts – it indicates that you should check out the last commit's version of Gemfile.lock and `bundle install` to resolve the conflict.

**Pull request merged**
